### DISCUSSION TITLE
"no module named odbserver" error

### DISCUSSION THREAD

**Discussion by MaximLutovinov** [boschresearch/pylife#58] [Q&A]
Hello,

I'm trying to load an FE mesh from an odb file, but I'm getting error "No module named odbserver". Would you have any recommendations on how I can resolve it? I've imported odbclient to my environment ("import odbclient as CL") and see "pylife-odbclient" item in list of packages for my environment, 

Thank you for any help in advance!

![pylife_no_odbserver_module_error](https://github.com/boschresearch/pylife/assets/154364083/0604c418-b638-4228-99c8-653f6b59ee4b)

**Comment by johannes-mueller** [comment]
It seems that you set as `python_env_path` the source code directory of the `odbserver`.

Have you set up an odbserver environment according to [the instructions](https://pylife.readthedocs.io/en/stable/tools/odbserver/index.html)? If yes, try again without setting the `python_env_path` argument. The `odbclient` should find it automatically.

**Comment by MaximLutovinov** [comment]
I unintentionally skipped the first half of the installation and was missing odbserver with python 2.7 environment... Fixed it!

The odbclient, however, didn't find python_env_path argument, so I used direct specification, which worked!

![pylife01](https://github.com/boschresearch/pylife/assets/154364083/6302fe6b-ac24-41b6-ad98-0aed45c49d65)

**Comment by MaximLutovinov** [comment]
I'm encountering an issue a bit further now, when trying to output stresses... any ideas what might be the problem?

![pylife02](https://github.com/boschresearch/pylife/assets/154364083/76e9c22c-388f-41e5-ad37-8f64b2ed7034)

**Comment by johannes-mueller** [comment]
That looks like you have a 2D model. If yes, please test with a 3D model.

**Reply by MaximLutovinov** [reply]
It is a 3D model, but it has beam and shell elements besides solids...

**Reply by johannes-mueller** [reply]
Ah, ok. These elements are not supported as of now. We probably should document this [here](https://pylife.readthedocs.io/en/stable/tools/odbclient/index.html#limited-functionality).

I would suggest to close this discussion here and open a new one in the "Ideas" section on what the odbclient should support. Would that be ok for you?

**Reply by MaximLutovinov** [reply]
Yes, of course! Thank you for all your answers!

**Comment by johannes-mueller** [comment]
Closing this thread.

Error message issue will be treated in #64.

For further discussions on how to deal with different elements refer #63
